src/components/noteListPage/NoteListPage.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { httpMethod, httpRequest } from '../../helpers/httpRequest';
+import { INote } from '../../interfaces/note';
+import Note from '../note/Note';
+import NoteCreation from '../noteCreation/NoteCreation';
+import Search from '../search/Search';
+
+type MyProps = {};
+type MyState = { notes: INote[] };
+
+export default class NoteListPage extends Component<MyProps, MyState> {
+
+  constructor(props: {}) {
+    super(props);
+    this.state = { notes: [] };
+    this.addNote = this.addNote.bind(this);
+    this.deleteNote = this.deleteNote.bind(this);
+    this.handleMarkAsDone = this.handleMarkAsDone.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleArchive = this.handleArchive.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadNotes();
+  }
+
+  render() {
+    return (
+      <div>
+      <nav className="navbar navbar-light bg-light">
+        <span className="navbar-brand mb-0 h1">My Notes 📝</span>
+        <Link to={'/archived'}>
+            <button className="btn btn-light">archived notes</button>
+        </Link>
+        <Search handleSearch={this.handleSearch}/>
+      </nav>
+      <div className="container">
+        <NoteCreation addNote={this.addNote}/>
+        {this.state.notes.length === 0 ?
+        <div>There is no notes</div> :
+        this.state.notes.map((note) =>
+        <Note
+          note={note}
+          key={note.id}
+          deleteNote={this.deleteNote}
+          handleMarkAsDone={this.handleMarkAsDone}
+          handleArchive={this.handleArchive}
+        />)}
+      </div>
+    </div>
+    );
+  }
+
+  private addNote(note: INote) {
+    httpRequest(httpMethod.post, 'notes', note)
+      .then((res) => res.json())
+      .then((n) => this.setState({ notes: [...this.state.notes, n] }))
+      .catch((er: Error) => console.error('Er', er));
+  }
+
+  private deleteNote(note: INote) {
+    httpRequest(httpMethod.delete, `notes/${note.id}`)
+      .then(() => this.setState({ notes: this.state.notes.filter((n) => n.id !== note.id) }))
+      .catch((er: Error) => console.error('Er', er));
+  }
+
+  private handleMarkAsDone(note: INote) {
+    httpRequest(httpMethod.patch, `notes/${note.id}`, { isDone: !note.isDone })
+      .then(() => {
+        const marked = this.state.notes.find((n) => n.id === note.id);
+        if (!marked) {
+          return;
+        }
+        const updatedNotes = this.state.notes.map((n) => n === marked ? { ...marked, isDone: !marked.isDone } : n);
+        this.setState({ notes: updatedNotes });
+      })
```

Comment:
There are at least two different ways to achieve the same result in a more elegant way:

```js

// Using a single intermediate variable instead of two

  private handleMarkAsDone(note: INote) {
    httpRequest(httpMethod.patch, `notes/${note.id}`, { isDone: !note.isDone })
      .then(() => {
        const updatedNotes = this.state.notes.map((n: INote) =>
          n.id === note.id ? { ...n, isDone: !n.isDone } : n)
        this.setState({ notes: updatedNotes });
      })
      .catch((er: Error) => console.error('Er', er));
  }


// setState can be a function if you need to derive from the current state:

  private handleArchive(note: INote) {
    httpRequest(httpMethod.patch, `notes/${note.id}`, { isArchived: true })
      .then(() => {
        this.setState((state: MyState) => ({
            notes: state.notes.filter((n: INote) => n.id !== note.id)
          })
        )
      })
      .catch((er: Error) => console.error('Er', er));
  }
```

Reply:
really really better!